controllers/chat/sendSignedMessage.js

```diff
@@ -101,18 +104,16 @@ const sendSignedMesage = async (req, res) => {
       if (isBlocked) {
         return res.status(403).json(responseError('This user does not want to receive messages'));
       }
-      // if (blockedIds.length > 0) {
-      //   for (const member of better_channel_member) {
-      //     if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
-      //       return res
-      //         .status(403)
-      //         .json(responseError('This user does not want to receive messages'));
-      //     }
-      //   }
-      // }
+
+      channelMember = better_channel_member.filter(
+        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+      );
+
+      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here filters out blocked users from the `channelMember` array and finds the sender's information. However, there's no error handling or checks to ensure that `channelMember` and `senderInfo` are not empty or undefined after these operations. Consider adding error handling or checks to handle these cases.
```javascript
channelMember = better_channel_member.filter(
  (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
);

if (!channelMember || channelMember.length === 0) {
  // Handle case when no channel members are found
}

senderInfo = better_channel_member.find((element) => element.user_id === req.userId);

if (!senderInfo) {
  // Handle case when sender info is not found
}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->